### DISCUSSION TITLE
Rotate EBSD Data

### DISCUSSION THREAD

**Discussion by mariuslueck** [mtex-toolbox/mtex#1910] [Ask Anything]
Hello all,

I am trying to rotate an EBSD measurement with MTEX. The reason is the following:
I have measured a single crystal perpendicular to the 001 direction, so I have cut it in some 100 direction. This gives me a larger area to measure dendrites.
Now the question arises whether I can convert the data with a suitable rotation of the orientations in the metaphorical sense as if I were looking at the 001 surface.
The next question would then be whether it is possible to calculate the deviation of the crystal orientation (here 001 growth direction) from the sample reference system (i.e. the rod centre axis) using this measurement.

Or does the data not allow such a calculation? We are dealing with nickel with a kfz structure. Data set is attached (saved as txt to upload it here but exportet as ang file) 

Many thanks
[EBSD_Test_100.ang.txt](https://github.com/mtex-toolbox/mtex/files/12194376/EBSD_Test_100.ang.txt)

**Comment by kilir** [comment]
Hi,
rotating the orientations is easily done e.g.:
```
ori = ebsd('Nickel').orientations;
rot = rotation.byAxisAngle(xvector,pi/2)
ori_rot = rot * ori
```

Rotating them inside the map would not make much sense I guess and rotating the map itself will give you a line.

Hope this helps.
Rüdiger

**Comment by mariuslueck** [comment]
Hello, thank you very much for the help. The rotation works very well. 

However, the question now is how I can conclude the deviation from the "rod axis" with the new Euler angles.
Normally Phi1 rotates around the Z axis, Theta around the X axis and Phi2 again around the Z axis. These rotations or the effects of the rotations about the angles and axes are now not the same with a 90 degree rotation of the orientations. 

Is the problem too specific to have a simple solution?

**Reply by kilir** [reply]
Hi,
just that I understand correctly, you have a reference ("rod axis") and you want to compute some measure of the deviation from it? Is that "rod axis" a direction in specimen coordinates or an orientation(fibre) or somethign complete different?
Cheers,
Rüdiger

**Reply by mariuslueck** [reply]
I'll try to be more specific:IIc I have a single crystal of nickel. Normally, the deviation of the 001 growth direction of the single crystal is determined by a measurement at the face. We measure the tilt of the dendrites from the rod axis of the single crystal. In the case of a measurement such as I have just described, the angle theta is the relevant angle because only the rotation around the X-axis and not the rotation around the Z-axis has an influence on the deviation of the dendrites from the rod axis.
Now, I did not make my EBSD measurement on the single crystal at the front side, but from a direction that is perpendicular to the 001 direction. For example, from the 100 direction. Accordingly, in this measurement the angle theta would indicate the deviation of the dendrite arms from my y-axis of the rod and not the deviation of my dendrite from the Z-axis of the rod. 

The figure tries to visualize the situation. In this ideal shematic figure the Z-axis is also the rod axis.

My idea was to "convert" the measurement so that I am, notionally, looking from the 001 direction even though I measured from 100. Actually, the orientations determined by EBSD give a clear position of the crystals. Therefore, it should theoretically be possible to calculate the deviation from 001 with a measurement from the 100 direction on the basis of the orientations.
Is that understandable?
![cubicebsd](https://github.com/mtex-toolbox/mtex/assets/140814577/ca8f22d7-e20c-4e66-83aa-e31a3c5bb1bb)

**Reply by mariuslueck** [reply]
Therefore the rotation was good start to simulate the change in direction of view. Now I need the the right mathematical operation to calculate the deviation with the new euler angles.

**Comment by kilir** [comment]
I think I may understand what you want to do and I do not think there's a need to rotate.

How about the following:
```
ebsd = loadEBSD_ang('EBSD_Test_100.ang','setting 2','convertEuler2spatialReferenceFrame')
cs = ebsd('Ni').CS

% a reference orientation 
ori0 = orientation.id(cs) % adapt to whatever is needed
h = Miller({1,0,0},{1,1,0},{1,1,1},cs)


plotPDF(ori0,h)
hold on
plotPDF(ebsd('Ni').orientations,h)
hold off

% angle between measurement and ref orientation
angle(ebsd('Ni').orientations,ori0)/degree

%% angle between reference specimen direction and a given crystal plane
refdir = vector3d(0,1,0);

% crystal plane of interest
h = Miller(1,0,0,cs);

% refernce direction in crystal coordinates
hm = inv(ebsd('Ni').orientations) .* refdir

% angle between specimen reference direction and crystal
angle(h,hm)/degree

```

Cheers,
Rüdiger

**Comment by mariuslueck** [comment]
Thank you for the code.
In the meantime I was able to validate my measurements with the OIM software, a rotation around the X-axis and the calculation of the deviation (as described by me) leads to an angle of approx. 10.7 degrees.

When I run the code I get about 14 degrees, with a little modification and the previous rotation I was able to validate the result.

Now it would be great if I could plot the resulting change in orientations as an IPF map.

Attached is the current code with which I get the same result as in the OIM software: (I don't know why but code function is not working)

% path to files
pname = 'Directory to file';

% which files to be imported
fname = [pname 'EBSD_Test_100.ang'];

ebsd = loadEBSD_ang(fname,'setting 2','convertEuler2spatialReferenceFrame')
CS = {... 
  'notIndexed',...
  crystalSymmetry('m-3m', [3.5 3.5 3.5], 'mineral', 'Ni')};

cs = ebsd('Ni').CS

% a reference orientation 
ori0 = orientation.id(cs) % adapt to whatever is needed
h = Miller({1,0,0},{1,1,0},{1,1,1},cs)

ori = ebsd.orientations;
rot = rotation.byAxisAngle(xvector,90*degree)
ori_rot = rot * ori

plotPDF(ori0,h)
hold on
plotPDF(ori_rot,h)
hold off

% angle between measurement and ref orientation
angle(ori_rot,ori0)/degree

%% angle between reference specimen direction and a given crystal plane
refdir = vector3d(1,0,0);

% crystal plane of interest
h = Miller(0,0,1,cs);

% refernce direction in crystal coordinates
hm = inv(ori_rot) .* refdir

% angle between specimen reference direction and crystal
angle(h,hm)/degree
